bindings/serialize-v8.cc

```diff
@@ -0,0 +1,208 @@
+
+  virtual std::string name() const override {
+    return *String::Utf8Value(node->name);
+  }
+
+  virtual std::string filename() const override {
+    return *String::Utf8Value(node->script_name);
+  }
+
+  virtual int64_t getFileID() const override { return node->script_id; }
+
+  virtual int64_t lineNumber() const override { return node->line_number; }
+
+  virtual int64_t columnNumber() const override { return node->column_number; }
+
+  virtual std::vector<Sample> samples(const std::deque<uint64_t>& stack,
+                                      Profile* profile) const override {
+    std::vector<Sample> samples;
+    for (size_t i = 0; i < node->allocations.size(); i++) {
+      AllocationProfile::Allocation allocation = node->allocations[i];
+      std::vector<Label> labels = {Label(profile->stringID("allocation"), 0,
+                                         allocation.size,
+                                         profile->stringID("bytes"))};
+      std::vector<int64_t> vals = {
+          allocation.count,
+          static_cast<int64_t>(allocation.size * allocation.count)};
+      Sample s = Sample({stack.begin(), stack.end()}, vals, labels);
+      samples.push_back(s);
+    }
+    return samples;
+  }
+};
+
+class TimeNode : public Node {
+ private:
+  const CpuProfileNode* node;
+  int samplingIntervalMicros;
+
+ public:
+  TimeNode(const CpuProfileNode* node, int samplingIntervalMicros)
+      : node(node), samplingIntervalMicros(samplingIntervalMicros) {}
+
+  virtual std::string name() const override {
+    return *String::Utf8Value(node->GetFunctionName());
+  }
+
+  virtual std::string filename() const override {
+    return *String::Utf8Value(node->GetScriptResourceName());
+  }
+
+  virtual int64_t getFileID() const override { return node->GetScriptId(); }
+
+  virtual int64_t lineNumber() const override { return node->GetLineNumber(); }
+
+  virtual int64_t columnNumber() const override {
+    return node->GetColumnNumber();
+  }
+
+  virtual std::vector<Sample> samples(const std::deque<uint64_t>& stack,
+                                      Profile* profile) const override {
+    std::vector<Sample> samples;
+    int64_t hitCount = node->GetHitCount();
+    std::vector<int64_t> vals = {hitCount, hitCount * samplingIntervalMicros};
+    Sample s = Sample({stack.begin(), stack.end()}, vals, std::vector<Label>());
+    samples.push_back(s);
+    return samples;
+  }
+};
+
+struct TimeEntry {
+  const CpuProfileNode* node;
+
+  // number of entries which would need to be removed from the stack after
+  // processing the node if the node were a leaf node.
+  int popCount;
+};
+
+std::unique_ptr<std::vector<char>> serializeTimeProfile(
+    CpuProfile* profileTree, int64_t samplingIntervalMicros,
+    int64_t startTimeNanos) {
+  int64_t durationNanos =
+      (profileTree->GetEndTime() - profileTree->GetStartTime()) * 1000;
+
+  Profile profile = Profile("wall", "microseconds", samplingIntervalMicros,
```

Comment:
`Profile profile("wall", ...)` to avoid a copy.

Reply:
done.